File: app/src/main/java/com/example/user/present/counter/MeasurementReceiver.java
package com.example.user.present.counter;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;

import timber.log.Timber;

public class MeasurementReceiver extends BroadcastReceiver {

    private static final String TAG = "MeasurementReceiver";

    public static final String ACTION_START_MEASUREMENT = "com.example.user.present.counter.ACTION_START_MEASUREMENT";
    public static final String ACTION_STOP_MEASUREMENT = "com.example.user.present.counter.ACTION_STOP_MEASUREMENT";

    @Override
    public void onReceive(Context context, Intent intent) {
        Timber.d("onReceive");

        final String action = intent.getAction();
        Timber.d("action = " + action);
        if (action.equals(ACTION_START_MEASUREMENT)) {
            Intent startServiceIntent = new Intent(context.getApplicationContext(), MeasurementService.class);
            context.startService(startServiceIntent);
        } else if (action.equals(ACTION_STOP_MEASUREMENT)) {
            Intent stopServiceIntent = new Intent(context.getApplicationContext(), MeasurementService.class);
            context.stopService(stopServiceIntent);
        }
    }

    @NonNull
    @Override
    public String toString() {
        return super.toString();
    }
}

File: app/src/main/java/com/example/user/present/counter/MeasurementReceiver.kt
package com.example.user.present.counter

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import timber.log.Timber

class MeasurementReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Timber.d("onReceive")
        val action = intent.action
        Timber.d("action = $action")
        if (action == ACTION_START_MEASUREMENT) {
            val startServiceIntent = Intent(context.applicationContext, MeasurementService::class.java)
            context.startService(startServiceIntent)
        } else if (action == ACTION_STOP_MEASUREMENT) {
            val stopServiceIntent = Intent(context.applicationContext, MeasurementService::class.java)
            context.stopService(stopServiceIntent)
        }
    }

    override fun toString(): String {
        return super.toString()
    }

    companion object {
        const val ACTION_START_MEASUREMENT = "com.example.user.present.counter.ACTION_START_MEASUREMENT"
        const val ACTION_STOP_MEASUREMENT = "com.example.user.present.counter.ACTION_STOP_MEASUREMENT"
    }
}